wren-ui/src/apollo/server/resolvers/projectResolver.ts

```diff
@@ -466,33 +466,133 @@ export class ProjectResolver {
     const modelColumns =
       await ctx.modelColumnRepository.findColumnsByModelIds(modelIds);
 
-    // Mapping with affected models and columns data into schame change
-    const mappingAffectedToSchemaChange = (changes: DataSourceSchema[]) => {
+    const modelRelationships = await ctx.relationRepository.findRelationInfoBy({
+      columnIds: modelColumns.map((column) => column.id),
+    });
+
+    const allCalculatedFields = modelColumns.filter(
+      (column) => column.isCalculated,
+    );
+
+    // Not only Mapping with affected models and columns data,
+    // But also finding out affected relationships and affected calculated fields into schema change
+    const mappingAffectedToSchemaChange = (
+      changes: DataSourceSchema[],
+      key: SchemaChangeType,
+    ) => {
       const affecteds = flatMap(changes, (change) => {
         const affectedModel = models.find(
           (model) => model.sourceTableName === change.name,
         );
-        return affectedModel
-          ? {
-              sourceTableName: change.name,
-              displayName: affectedModel.displayName,
-              columns: flatMap(change.columns, (column) => {
-                const affectedColumn = modelColumns.find(
-                  (modelColumn) =>
-                    modelColumn.sourceColumnName === column.name &&
-                    modelColumn.modelId === affectedModel.id,
+
+        if (!affectedModel) return [];
+
```

Comment:
Could we add a minor comment here explaining why we need `selfModelCalculatedFields` ?